### PR TITLE
Icos citation

### PR DESCRIPTION
**Pull request by KarolinaPntzt**
Hi!
I have now updated sparqls.py in pytools/icoscp/sparql/ to include sparql query get_icos_citation().
Changes have also been made to the ICOS Jupyter Notebooks (icos_obs, icos_stilt, icos_ccgcrv), i.e deleted old function get_icos_citation(dobj), to get the citation string  by using get_icos_citation() from icoscp.

### CODE REVIEW

**Comment by claudiodonofrio** [comment]
Hi I have only checked the sparqls.py.
I suggest to modify it slightly. For consistency and to make it more foolproof there is a function at the very top of the file __dobjUrl__(dobj) which makes sure that you can provide the PID  or the full url. Hence i would start in your function with dataObject = __dobjUrl__(dataObject)
and probably you could delete the line with:
#Define URL:    url = 'https://meta.icos-cp.eu/sparql'
it looks not used in the function

**Comment by claudiodonofrio** [comment]
ahhh github. it has removed the double underscore in the function name.........

File: pytools/icoscp/sparql/sparqls.py [hunk 1]
```diff
@@ -724,3 +724,39 @@ def icos_hist_sparql():
...
+                      a unique identifier of every separate ICOS dataset.
+    
+    Input parameters: Data Object ID (var_name: "dataObject", var_type: String)
+    
+    Output:           Citation (var_type: String)
+
+    """
+    
+    #Define URL:
+    url = 'https://meta.icos-cp.eu/sparql'
```
Comment (claudiodonofrio) [inline pytools/icoscp/sparql/sparqls.py]: delete, not used

File: pytools/icoscp/sparql/sparqls.py [hunk 2]
```diff
@@ -724,3 +724,39 @@ def icos_hist_sparql():
...
+
```
Comment (claudiodonofrio) [inline pytools/icoscp/sparql/sparqls.py:750]: # check input variable 
dataObject = __dobjUrl__(dataObject)

Comment (claudiodonofrio) [reply pytools/icoscp/sparql/sparqls.py:750]: and again the double underscore has been removed from github.

**Review by claudiodonofrio — CHANGES_REQUESTED** [review]
see comments in sparqls.py

File: pytools/icoscp/sparql/sparqls.py (hunk 1 above)
Comment (ukarst) [reply pytools/icoscp/sparql/sparqls.py]: I agree, this is not needed

File: pytools/icoscp/sparql/sparqls.py (hunk 2 above)
Comment (ukarst) [reply pytools/icoscp/sparql/sparqls.py:750]: ok, to check if input variable is PID or url 
BUT in this case '<' and '>' need to be removed when composing the query string.

**Comment by ukarst** [comment]
I also only checked the get_icos_citation() part
see my comments to Claudio's comments
overall, I support implementing the change

**Comment by KarolinaPntzt** [comment]
I have now implemented the requested changes in get_icos_citation() as well as added some new features to ccg_icos_tools.ipynb (i.e. solving missing sampling heights issue, solving mutliple sampling heights issue, adding interactive plot for yearly detrended values and adding sepparate plot for std dev.).

**Pull request merged**